### DISCUSSION TITLE
roundNumber is amibguous

### DISCUSSION THREAD

**Discussion by zoidyzoidzoid** [saul/demofile#226] [Questions] [answered]
Thanks for this awesome library, and all your great work on it. I hope I can contribute to it in the future.

`roundNumber` sounds like it'll be the round number for the round currently in progress, but it changes when the score changes, before the round actually ends. The field's name of [`m_totalRoundsPlayed`](https://github.com/saul/demofile/blob/master/entities/gamerules.js#L19)  sounds more accurate, and if people wanna keep track of the round currently in progress then `roundNumber` would probably be `m_totalRoundsPlayed` + 1, except between the round ending and the round actually ending.

`round_end`: when the bomb explodes / last player dies and the score changes, but folks can still move around for some amount of time to save weapons or whatever.
`round_officially_ended`: when players finally "respawn" / move back to their spawns.

**Comment by saul** [comment, marked as answer]
Maybe a better idea would be to store m_totalRoundsPlayed on round_start, and then `roundNumber` would return that stored number +1? Seems a bit more accurate.

I’m loathe to add event handlers and state though as everyone who uses the library is paying the cost for a feature that they may not use.

**Comment by zoidyzoidzoid** [comment]
Yeah, maybe it's better to just document it somewhere or include it in an example? I'm happy to provide one. 

I've worked with some other parsers that do try store state themselves and include their own events, and have enjoyed and learnt a lot from how this parser does it.
